pkg/util/arangod/agency_health.go

```diff
@@ -0,0 +1,98 @@
+//
+// DISCLAIMER
+//
+// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Ewout Prangsma
+//
+
+package arangod
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+)
+
+const (
+	maxAgentResponseTime = time.Second * 10
+)
+
+// agentStatus is a helper structure used in AreAgentsHealthy.
+type agentStatus struct {
+	IsLeader       bool
+	LeaderEndpoint string
+	IsResponding   bool
+}
+
+// AreAgentsHealthy performs a health check on all given agents.
+// Of the given agents, 1 must respond as leader and all others must redirect to the leader.
+// The function returns nil when all agents are healthy or an error when something is wrong.
+func AreAgentsHealthy(ctx context.Context, clients []Agency) error {
+	wg := sync.WaitGroup{}
+	invalidKey := []string{"does-not-exists-149e97e8-4b81-5664-a8a8-9ba93881d64c"}
```

Comment:
Just for purpose of beauty:  drop the `s` in `exists`

Reply:
done